### DISCUSSION TITLE
Use Nitter to bypass API limits

### DISCUSSION THREAD

**Discussion by robsonsobral** [robertodoering/harpy#811] [Ideas]
![Screenshot_20221019-212458.png](https://user-images.githubusercontent.com/931642/196828733-3e10a860-441a-4336-b657-132df8a1bba3.png)

TwidereX uses Nitter to bypass API limits. Could Harpy do the same?

**Comment by Patrick-web** [comment]
I was also thinking of away to go around this and my suggestion was using a hidden webview that is logged into the user's twitter. The app can scrape the webview for content and display to the user.

**Reply by robsonsobral** [reply]
Too error prone, don't you think? Any Twitter change could break everything.

**Comment by yoshimo** [comment]
Is this still an issue with the twitter v2 api?

**Reply by robertodoering** [reply]
Yes, in some cases the limits are even worse compared to the v1.1 endpoints.

**Comment by robertodoering** [comment]
The limitations that you run into with harpy frequently are the app-wide api limits such as liking a tweet.
To my knowledge, Nitter can't be used to like, retweet or interact with the platform in any way. The whole point seems to be not to talk to Twitter and instead to just fetch and display information.
So for harpy this won't be very useful.

**Reply by robsonsobral** [reply]
I thought the limits are consolidated. So, to fetch tweets from Nitter would save requests to be used on other things.

😞

**Reply by robertodoering** [reply]
Every endpoint has its own limits, most of which are user-based (e.g. every user can refresh their timeline 15 times every 15 minutes).
Some, however, are app-wide (such as liking a tweet) and shared by every user (e.g. 1000 likes every 24 hours) 😞

**Reply by yoshimo** [reply]
I think before v2 api it could be used to collect threads in a less limited way, but that benefit is small.

**Reply by robertodoering** [reply]
Yeah that is true, we do get access to some features with it that we don't have right now - however we might as well start migrating to v2 before using a third party api.

**Reply by robsonsobral** [reply]
From a user point of view, what would be better to have sooner? Threads or... I don't know what else.

**Comment by Patrick-web** [comment]
I have a controversial and risky suggestion. How about reverse engineering their private api. Get the API tokens once the user logs into their account via the webview and then use the tokens to perform actual api requests.
